gormschema/gorm.go

```diff
@@ -73,12 +74,79 @@ func (l *Loader) Load(models ...any) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	if err := db.AutoMigrate(models...); err != nil {
-		return "", err
+	db.Config.DisableForeignKeyConstraintWhenMigrating = true
+	err = db.AutoMigrate(models...)
+	if !l.config.DisableForeignKeyConstraintWhenMigrating {
```

Comment:
is it possible to invoke the new logic only if cycles are detected?

Reply:
Not that I'm aware of, I can't find any clear error from AutoMigrate that we can use. 
We could probably search the graph, but that would require some more research